packages/sources/wootrade/src/endpoint/crypto.ts

```diff
@@ -0,0 +1,37 @@
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+import { Config, ExecuteWithConfig, InputParameters } from '@chainlink/types'
+import { NAME as AdapterName } from '../config'
+
+export const supportedEndpoints = ['crypto', 'ticker']
+
+const customError = (data: any) => data.Response === 'Error'
+
+export const inputParameters: InputParameters = {
+  base: ['base', 'from', 'coin'],
+  quote: ['quote', 'to', 'market'],
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  let base = validator.overrideSymbol(AdapterName)
+  if (Array.isArray(base)) base = base[0]
+  const quote = validator.validated.data.quote
+  const symbol = `SPOT_${base.toUpperCase()}_${quote.toUpperCase()}`
+  const limit = '1'
+  const url = `/v1/public/market_trades/`
+
+  const params = {
+    symbol,
+    limit
+  }
+
+  const options = { ...config.api, params, url }
+  const response = await Requester.request(options, customError)
```

Comment:
`Requester.request` also supports taking a type for the response. Could we add the type of the response schema, and also export it? Here's an example: https://github.com/smartcontractkit/external-adapters-js/blob/develop/packages/sources/coinmetrics/src/endpoint/price.ts